models/validators/UniqueValidator.cfc

```diff
@@ -2,64 +2,113 @@
  * Copyright since 2020 by Ortus Solutions, Corp
  * www.ortussolutions.com
  * ---
- * This validator checks the database according to validation data for field uniqueness
- * - table : The table name to seek
- * - column : The column to evaluate for uniqueness or defaults to the name of the field
+ * This validator validates if a value is unique in a database
  */
-component
-	extends  ="BaseValidator"
-	accessors="true"
-	singleton
-{
+component accessors="true" singleton {
+
+	property name="name";
+	property name="res" 				inject="model:resourceService@cbi18n";
+    property name="moduleService" inject="coldbox:moduleService";
+    property name="wirebox" inject="wirebox";
 
 	/**
 	 * Constructor
 	 */
-	UniqueValidator function init(){
-		variables.name = "Unique";
+	QUniqueValidator function init() {
```

Comment:
Why `QUniqueValidator`?

Reply:
yes, stupid, copy-paste error. I had two different unique validators in my code, one based on qb, the other on queries. One of them had q in it, but we don't need it anymore. Only query based validator here